captum/attr/_core/feature_ablation.py

```diff
@@ -0,0 +1,329 @@
+    ):
+        num_features = torch.max(input_mask).item() + 1
+        num_examples = inputs[0].shape[0]
+        batch_size = (
+            num_features
+            if internal_batch_size is None
+            else min(internal_batch_size // num_examples, num_features)
+        )
+        baseline = baselines[i] if isinstance(baselines, tuple) else baselines
+        if isinstance(baseline, torch.Tensor):
+            baseline = baseline.reshape((1,) + baseline.shape)
+
+        # Repeat features and additional args for batch size.
+        all_features_repeated = [
+            torch.cat([inputs[j]] * batch_size, dim=0) for j in range(len(inputs))
+        ]
+        additional_args_repeated = (
+            _expand_additional_forward_args(additional_args, batch_size)
+            if additional_args is not None
+            else None
+        )
+
+        num_features_processed = torch.min(input_mask).item()
+        while num_features_processed < num_features:
+            current_batch = min(batch_size, num_features - num_features_processed)
+
+            # Store appropriate inputs and additional args based on batch size.
+            if current_batch != batch_size:
+                current_features = [
+                    tensor[0 : current_batch * num_examples]
+                    for tensor in all_features_repeated
+                ]
+                current_additional_args = (
+                    _expand_additional_forward_args(additional_args, current_batch)
+                    if additional_args is not None
+                    else None
+                )
+            else:
+                current_features = all_features_repeated
+                current_additional_args = additional_args_repeated
+
+            # Store existing tensor before modifying
+            original_tensor = current_features[i]
+            # Construct ablated batch for features in range num_features_processed
+            # to num_features_processed + current_batch and return mask with
+            # same size as ablated batch.
+            ablated_features, current_mask = self._construct_ablated_input(
+                current_features[i].reshape(
+                    (current_batch, num_examples) + current_features[i].shape[1:]
+                ),
+                input_mask,
+                baseline,
+                num_features_processed,
+                num_features_processed + current_batch,
+            )
+            current_features[i] = ablated_features.reshape(
+                (-1,) + ablated_features.shape[2:]
+            )
+            yield tuple(current_features), current_additional_args, current_mask
+            # Replace existing tensor at index i.
+            current_features[i] = original_tensor
+            num_features_processed += current_batch
+
+    def _construct_ablated_input(
+        self, feature_tensor, input_mask, baseline, start_feature, end_feature
+    ):
+        r"""
```

Comment:
@miguelmartin75 I think you should be able to implement shuffling on top of this by just overriding this function to shuffle each batch instead of replacing with the baseline.